tasks

esthete014 opened the pull request:
created class
prisvaivanie i vivod

esthete014 committed:
18.02.2022 5:36 created class and 1, 5 points

esthete014 committed:
Create .gitkeep

esthete014 committed:
Delete .gitkeep

esthete014 committed:
Update README.md

esthete014 committed:
Update README.md

esthete014 commented:
@UsovaMA
i didn't fork, cause it looks worse than it(
**mini instruction:**

> 1) go to repository "OOP"(i pinned it)
> 2) switch branch to "in-progress"
> 3) go to folder "CHECK_THIS"
> 4) choose task which u need to check


_I hope you feel comfortable with this)
BUT if it hard to check, i can fork ur repository_


**also I would like to propose to do:**
in [table](https://docs.google.com/spreadsheets/d/1mhWoqNYzbAO3Qelq1jaRQXjDq1HvilVeSgbpnKRbdB0/edit#gid=1301536656) create url in cells to git for everyone of group

========================

**_на русском:_**

я не форкнул, потому что это выглядит хуже, чем это(
**мини-инструкция:**

> 1) перейти в репозиторий "ООП" (я его закрепил)
> 2) переключить ветку на "in-progress"
> 3) перейти в папку "CHECK_THIS"
> 4) выберите task, который вам нужно проверить


_Надеюсь, вам это удобно)
НО, если это сложно так проверять, я могу сделать как раньше через fork_


**также я хотел предложить сделать:**
в [таблице](https://docs.google.com/spreadsheets/d/1mhWoqNYzbAO3Qelq1jaRQXjDq1HvilVeSgbpnKRbdB0/edit#gid=1301536656) создайте URL-адреса на git в ячейках для каждого из группы

UsovaMA commented on CHECK_THIS/task1/Header.h:
```diff
@@ -0,0 +1,167 @@
+		m_rub = 0;
+		m_cent = 0;
+		std::cout << "RUBLES: " << m_rub << std::endl;
+		std::cout << "CENTS: " << m_cent << "\n" << std::endl;
```
Плохое решение. Не надо в конструкторах, назначение которых только в создании объектов, лепить печать на экран. Более того скажу, что конструкторы вызываются не только по указке, когда мы передаём в функцию параметры, там также могут вызваться фоново конструкторы и увидишь в итоге какой-то бред на экране с кучу напечатанной информации, хотя хотел, например, сложить два объекта.

esthete014 replied:
done

UsovaMA commented on CHECK_THIS/task1/Header.h:
```diff
@@ -0,0 +1,167 @@
+		//Money _name;
+		//m_rub = r;
+		//m_cent = c;
+	}
```
Переименовала бы этот метод в inputKeyboard(); то есть ввод с клавиатуры и оставила бы только один. Использовать Сшый код в C++ моветон. Нужно выбирать с чем работать, но сейчас у вас нет выбора, так как мы учимся именно С++.

esthete014 replied:
done

esthete014 committed:
Add files via upload

UsovaMA commented on CHECK_THIS/task1/Header.h:
```diff
@@ -0,0 +1,167 @@
+			//end of while for work
+		}
+		//end of void work
+	}
```
Нарушено всё, что можно нарушить в ООП.
1) приложение для работы с классом реализуется в main(), либо рядом с ним в отдельной функции.
2) класс - самодостаточная единица, внутри него должны быть реализованы методы, которые могут быть применены к указанному объекту.

```
class Name {
  type fieldName;
public:
  Name();
  Name(type _fn);
  Name(const Name& n);

  type1 method1(type_1 t1, type_2 t2);
  type2 method2(type_1 t1, type_2 t2);
  //...
};

int main() {
  Name obj;
  //...
  switch (chosen_method) {
    case 1:
      // например, ввод p1 и p2
      // ...
      obj.method1(p1,p2);
      break;
    case 2:
      //...
  }
}
```

esthete014 replied:
done

UsovaMA reviewed (commented):
@esthete014 

> также я хотел предложить сделать:
> в [таблице](https://docs.google.com/spreadsheets/d/1mhWoqNYzbAO3Qelq1jaRQXjDq1HvilVeSgbpnKRbdB0/edit#gid=1301536656) создайте URL-адреса на git в ячейках для каждого из группы

в предложении отказано :)

К выкладыванию материалов претензий не имею, вы уже ребята взрослые, можете не гите сами грамотно свою работу осуществлять.

А вот по реализации замечаний много. 

![image](https://user-images.githubusercontent.com/22386453/155715777-ecdd2f18-632f-4f13-bb16-68edd3ddc274.png)

Главное замечание - не пихать в класс то, что относится к использованию класса (пользовательский интерфейс). **Класс - это логика объекта!!!** Никаких пользовательских интерфейсов. 

Методов по задаче мало сделано
+ **перегрузки** операторов жду (ссылки на материалы по теле закреплены в таблице в графе от 24.02.2022)

Зачтено авансом, поправить.

esthete014 commented:
> @esthete014
> 
> > также я хотел предложить сделать:
> > в [таблице](https://docs.google.com/spreadsheets/d/1mhWoqNYzbAO3Qelq1jaRQXjDq1HvilVeSgbpnKRbdB0/edit#gid=1301536656) создайте URL-адреса на git в ячейках для каждого из группы
> 
> в предложении отказано :)
> 
> К выкладыванию материалов претензий не имею, вы уже ребята взрослые, можете не гите сами грамотно свою работу осуществлять.
> 
> А вот по реализации замечаний много.
> 
> ![image](https://user-images.githubusercontent.com/22386453/155715777-ecdd2f18-632f-4f13-bb16-68edd3ddc274.png)
> 
> Главное замечание - не пихать в класс то, что относится к использованию класса (пользовательский интерфейс). **Класс - это логика объекта!!!** Никаких пользовательских интерфейсов.
> 
> Методов по задаче мало сделано
> 
> * **перегрузки** операторов жду (ссылки на материалы по теле закреплены в таблице в графе от 24.02.2022)
> 
> Зачтено авансом, поправить.

Cпасибо)
очень круто, нам с Максом мем понравился :з

я уже на почту писал, что я болею, так что я доделаю все, а некоторые замечания я и так понимал, что в будущем надо исправить

спасибки😊

esthete014 committed:
Add files via upload

esthete014 commented:
методы и перегрузку добавлю позже

esthete014 committed:
add class menu with  interface

esthete014 committed:
Delete Header.h

esthete014 committed:
отчаяние(

Здравствуйте, я просидел, не выкладывая, с этим проектом очень долго. Вначале, мне показалось это достаточно простой задачей, поэтому я даже усложнил себе задание добавочным вариантом + выводом/чтением/записью и другими действиями с файлами(их я кстати реализовал), НО я очень много потратил времени из-за непродуманности(чаще всего несоответствии типов данных и ограниченности функций того же вектора таких как обращение по элементу, которое на работает, возможно, оно будет работать при перезапуске проекта и много других ошибок, которые у меня решаются так). Мое начало с полным энтузиазма было подбито. Так что, надеюсь, когда я отдохну, я смогу переделать его.

esthete014 committed:
Delete .gitkeep

esthete014 committed:
slojenie matrix, vse rabotaet

esthete014 committed:
очень много сделал) 

вроде, даже, все отлично работает(я тестил все что мог, но мозги уже кипят)
не хватает только 3 и 7 пункта, так как я даже не понимаю, что это, а так даже с верхом сделано

color добавил, потому что красиво😊

esthete014 commented:
> 
> в предложении отказано :)
> 

Я понял, я просто имел ввиду, если вам не удобно так смотреть, когда репозиторий не форкнут, но да, разумно не делать ссылки, по понятным причинам

esthete014 committed:
умножение на скаляр

Если я правильно понял, то умножение матрицы на скаляр это просто умножение матрицы на число, обновил меню, чтобы можно было выйти в выбор варианта, а вариант с HEX, надеюсь скоро, доделаю

esthete014 committed:
Create .gitkeep

esthete014 committed:
Add files via upload

esthete014 committed:
Delete .gitkeep

esthete014 committed:
Create ReadMe

esthete014 committed:
update var5

exit, delete(enter and all), create(enter and random) with files, addition, compare
